smartapp-core/src/main/java/com/smartthings/sdk/smartapp/core/extensions/contextstore/DefaultInstalledAppContextStore.java

```diff
@@ -0,0 +1,29 @@
+package com.smartthings.sdk.smartapp.core.extensions.contextstore;
+
+import com.smartthings.sdk.smartapp.core.RequestPreprocessor;
+import com.smartthings.sdk.smartapp.core.models.ExecutionRequest;
+
+
+/**
+ * DefaultInstalledAppContextStore
+ */
+public interface DefaultInstalledAppContextStore
+        extends InstalledAppContextStore<DefaultInstalledAppContext>, RequestPreprocessor {
+    @Override
+    default void process(ExecutionRequest executionRequest) {
+        switch (executionRequest.getLifecycle()) {
+            case INSTALL:
+                add(DefaultInstalledAppContext.from(executionRequest.getInstallData()));
+                break;
+            case UPDATE:
+                update(DefaultInstalledAppContext.from(executionRequest.getUpdateData()));
+                break;
+            case UNINSTALL:
+                String installedAppId = executionRequest.getUninstallData().getInstalledApp().getInstalledAppId();
+                remove(installedAppId);
+                break;
+            default:
```

Comment:
In the future, let's support `CONFIGURATION` as well – users will inevitably want to persist values _between pages_.

Reply:
That's a good idea though we will need to be very careful with it and probably store the temporary data "off to the side" (e.g. in a separate field of the object) in the case of configuration from an update so we don't lose anything if case they hit cancel.